akka-http-core/src/test/scala/akka/http/impl/engine/parsing/HttpHeaderParserSpec.scala

```diff
@@ -186,6 +187,19 @@ abstract class HttpHeaderParserSpec(mode: String, newLine: String) extends WordS
         "HTTP header value exceeds the configured limit of 1000 characters"
     }
 
+    "produce an error when illegalResponseHeaderValueProcessingMode set to Error and a header contains illegal value" in new TestSetup(testSetupMode = TestSetupMode.Default, createParserSettings(system, illegalResponseHeaderValueProcessingMode = IllegalResponseHeaderValueProcessingMode.Error)) {
+      the[IllegalHeaderException] thrownBy parseLine(s"Server: something; something${newLine}x")
+    }
+
+    "continue parsing even if a header contains an invalid value when illegalResponseHeaderValueProcessingMode is set to Warn" in new TestSetup(testSetupMode = TestSetupMode.Default, createParserSettings(system, illegalResponseHeaderValueProcessingMode = IllegalResponseHeaderValueProcessingMode.Warn)) {
+      noException should be thrownBy parseLine(s"Server: something; something${newLine}x")
+    }
+
+    // TODO this test is almost identical to the test above, but has a different setup - what do?
+    "continue parsing even if a header contains an invalid value when illegalResponseHeaderValueProcessingMode is set to Ignore" in new TestSetup(testSetupMode = TestSetupMode.Default, createParserSettings(system, illegalResponseHeaderValueProcessingMode = IllegalResponseHeaderValueProcessingMode.Ignore)) {
+      noException should be thrownBy parseLine(s"Server: something; something${newLine}x")
+    }
+
```

Comment:
What about the tests where they're identical other than the test setup? Is there a way to perform the same test with two different setups?